api/v1alpha1/regexp_test.go

```diff
@@ -0,0 +1,347 @@
+package v1alpha1_test
+
+import (
+	"fmt"
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"io/fs"
+	"regexp"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func Test_Pattern_Branch(t *testing.T) {
+	testCases := map[string]bool{
+		"":             false,
+		"foo/bar":      true,
+		"foo.bar":      true,
+		"release-0.58": true,
+		"/foo":         false,
+		"foo/":         false,
+		"foo//bar":     true,
+		".foo":         false,
+		"foo.":         false,
+		"foo..bar":     true,
+	}
+
+	expression, err := findExpression("Branch")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for tt, expected := range testCases {
+		t.Run(tt, func(t *testing.T) {
+			require.Equal(t, expected, expression.MatchString(tt))
+		})
+	}
+}
+
+func Test_Pattern_Digest(t *testing.T) {
+	testCases := map[string]bool{
+		"":                        false,
+		"sha256:":                 false,
+		"sha256:1234567890abcdef": true,
+		":1234567890abcdef":       false,
+		"sha256:xyz":              false,
+		"xyz:1234567890abcdef":    true,
+		"sha256:1234567890abcdef1234567890abcdef1234567890abcdef1234567890abcdef": true,
+	}
+
+	expression, err := findExpression("Digest")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for tt, expected := range testCases {
+		t.Run(tt, func(t *testing.T) {
+			require.Equal(t, expected, expression.MatchString(tt))
+		})
+	}
+}
+
+func Test_Pattern_Duration(t *testing.T) {
+	testCases := map[string]bool{
+		"":  false,
+		"s": false,
+		".": false,
+		// simple cases
+		"10s": true,
+		"20h": true,
+		// Decimal cases
+		"1.5s": true,
+		".5s":  false,
+		"0.5s": true,
+		"1.5":  false,
+		"1.5h": true,
+		"1.5m": true,
+		"1,4s": false,
+		// edge cases
+		"1":    false,
+		"0":    false,
+		"-1":   false,
+		"-1s":  false,
+		"+33s": false,
+	}
+
+	expression, err := findExpression("Duration")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for tt, expected := range testCases {
+		t.Run(tt, func(t *testing.T) {
+			require.Equal(t, expected, expression.MatchString(tt))
+		})
+	}
+}
+
+func Test_Pattern_HelmRepoURL(t *testing.T) {
+	testCases := map[string]bool{}
```

Comment:
I couldn't find any PR that mentions any test cases for helm chart URLs.  They could be added here or later.  The test _still has value_ because it ensures that all uses of HelmRepoURL are in sync.